From 5af23db759209666a98044df2fa8c4c4430f36f1 Mon Sep 17 00:00:00 2001
From: Dragos Bogdan <dragos.bogdan@analog.com>
Date: Thu, 14 Jun 2018 11:07:55 +0300
Subject: [PATCH 15/25] dma: axi-dmac: Enable FLAG_LAST independent of
 FLAG_CYCLIC

An example when these two bits should be enabled simultaneously is
where the controller is used as a Video DMA and the TLAST signal acts
as an "end of frame" signal.

Signed-off-by: Dragos Bogdan <dragos.bogdan@analog.com>
(cherry picked from commit a288f0fe3d120bb04f15ed4b699a4eb1d4acf7fb)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/dma/dma-axi-dmac.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/dma/dma-axi-dmac.c b/drivers/dma/dma-axi-dmac.c
index 9328407cf204..a6de75b6d954 100644
--- a/drivers/dma/dma-axi-dmac.c
+++ b/drivers/dma/dma-axi-dmac.c
@@ -232,8 +232,8 @@ static void axi_dmac_start_transfer(struct axi_dmac_chan *chan)
 			desc->num_submitted = 0; /* Start again */
 		} else {
 			chan->next_desc = NULL;
-			flags |= AXI_DMAC_FLAG_LAST;
 		}
+		flags |= AXI_DMAC_FLAG_LAST;
 	} else {
 		chan->next_desc = desc;
 	}
-- 
2.17.0

